### PR TITLE
[cxx-interop] Provide a support header for common attributes, etc.

### PR DESCRIPTION
**Pull request by zoecarver**

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci Please Build Toolchain macOS Platform

File: lib/ClangImporter/cxx-interop.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#define SELF_CONTAINED __attribute__((swift_attr("import_owned")))
+#define SAFE_TO_IMPORT __attribute__((swift_attr("import_unsafe")))
+
+#define _CXX_INTEROP_STRINGIFY(_x) #_x
+#define REFERENCE(_retain, _release)                                           \
+  __attribute__((swift_attr("import_reference")))                              \
+  __attribute__((swift_attr(_CXX_INTEROP_STRINGIFY(retain:_retain))))          \
+  __attribute__((swift_attr(_CXX_INTEROP_STRINGIFY(release:_release))))
+
+#define SWIFT_NAME(_name) __attribute__((swift_name(#_name)))
```
Comment (egorzhdan) [inline lib/ClangImporter/cxx-interop.h:29]: I don't think `SWIFT_NAME` belongs here: it isn't specific to C++ interop, it is also used for C/ObjC interop. On Darwin there's an existing macro for this (`NS_SWIFT_NAME`), is this a replacement for `NS_SWIFT_NAME`?

Comment (zoecarver) [reply lib/ClangImporter/cxx-interop.h:29]: I was thinking more about non-darwin platforms or users that don't have access to or don't want to use Foundation (the place this macro is currently provided).

**Comment by zoecarver** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by zoecarver** [comment]
@swift-ci please test

**Pull request merged**